app/src/main/java/org/openobservatory/ooniprobe/common/MKCollectorResubmitTask.java

```diff
@@ -73,16 +73,16 @@ private static void perform(Context c, Measurement m) throws IOException {
 		List<Measurement> measurements = SQLite.select().from(Measurement.class).where(where.toArray(new SQLOperator[0])).queryList();
 		for (int i = 0; i < measurements.size(); i++) {
 			A activity = getActivity();
-			if (activity != null) {
-				String paramOfParam = activity.getString(R.string.paramOfParam, Integer.toString(i + 1), Integer.toString(measurements.size()));
-				publishProgress(activity.getString(R.string.Modal_ResultsNotUploaded_Uploading, paramOfParam));
-				Measurement m = measurements.get(i);
-				m.result.load();
-				try {
-					perform(activity, m);
-				} catch (IOException e) {
-					e.printStackTrace();
-				}
+			if (activity == null)
+				break;
```

Comment:
This is not equivalent to how it was before. To be equivalent you should do `continue` when it's null.

Reply:
i know, in that case break is better

Reply:
Why? In which cases can the activity be null?

I now have more question then:
* Why are we creating multiple instances of the activity on every cycle loop and not just creating a single one at the beginning of the loop?
* In which cases will the activity be null on an iteration of the loop? Should we perhaps not throw an exception in that case?

Reply:
we do not create multiple instances, only geting multiple times same instance.
we do that because use can close app during process, and in this way app will closed and not killed

Reply:
> we do not create multiple instances, only geting multiple times same instance.

Ah yes that's a good point.

> we do that because use can close app during process, and in this way app will closed and not killed

 Can you elaborate on this and better explain what you mean by this?

In which cases will the activity be null?

Reply:
When user close app during upload

Reply:
And by close you mean the user puts the app in background or they actually force close it?

What happens to measurements that are being uploaded? Is there a possibility of a race condition or of the DB ending up in an inconsistent state?

What happens if the user re-opens the app, are they able to resume the upload?

Reply:
Force close,

I made each measurement upload atomic, so on second run start on remaining measurements

Reply:
so i can resolve this conversation ?